airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferDequeue.java

```diff
@@ -74,7 +74,14 @@ public MemoryAwareMessageBatch take(final StreamDescriptor streamDescriptor, fin
         }
       }
 
-      queue.addMaxMemory(-bytesRead.get());
+      if (queue.isEmpty()) {
+        buffers.remove(streamDescriptor);
+        // free the remainder, leaving the read bytes to be freed by the batch flush
+        final var slop = queue.getMaxMemoryUsage() - bytesRead.get();
```

Comment:
Is this right? I feel that the maxMemory usage is a fix number.

Reply:
We add to maxMemory as we allocate [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferEnqueue.java#L60). From what I can tell `maxMemory` represents the total memory allocated for the buffer